File: Excel/500-599/537/537 Challenge.py
import pandas as pd
import numpy as np
from itertools import combinations

path = "537 Minimum Product for Triplet.xlsx"
input = pd.read_excel(path, usecols="A:F")
test = pd.read_excel(path, usecols="G")

output = input.copy()
output['min_product'] = input.apply(lambda row: min(np.prod(c) for c in combinations(row, 3)), axis=1).astype(np.int64)

print(output["min_product"].equals(test["Answer Expected"])) # True

File: Excel/500-599/537/537 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/537 Minimum Product for Triplet.xlsx"
input = read_excel(path, range = "A1:F10")
test  = read_excel(path, range = "G1:G10")

output = input %>%
  mutate(min_product = pmap_dbl(., ~ min(combn(c(...), 3, prod))))

identical(test$`Answer Expected`, output$min_product) 
#> [1] TRUE
